postreise/analyze/time.py

```diff
@@ -1,20 +1,66 @@
+import re
+
 import pandas as pd
 import pytz
 
 from postreise.analyze.check import _check_date_range_in_time_series, _check_time_series
 
 
-def slice_time_series(ts, start, end):
+def is_24_hour_format(time):
+    """Check if the input string is in 24-hour format
+
+    :param str time: input string
+    :return: (*bool*) -- the input string is in 24-hour format or not
+    """
+    regex = "^([01]?[0-9]|2[0-3]):[0-5][0-9]$"
+    p = re.compile(regex)
+    m = re.search(p, time)
+    return m is not None
+
+
+def slice_time_series(ts, start, end, between_time=None, dayofweek=None):
     """Slice a time series.
 
     :param pandas.DataFrame/pandas.Series ts: time series to slice.
     :param pandas.Timestamp/numpy.datetime64/datetime.datetime start: start date.
     :param pandas.Timestamp/numpy.datetime64/datetime.datetime end: end date.
+    :param list between_time: specify the start hour and end hour of each day
+        inclusively, default to None, which includes every hour of a day. Note that if
+        the end hour is set before the start hour, the complementary hours of a day are
+        picked.
+    :param set dayofweek: specify the interest days of week, which is a subset of
+        integers in [0, 7], default to None, which includes every day of a week.
```

Comment:
[1, 7]? What is the first and last days of the week, i.e., is Sunday or Monday first? and is Saturday or Sunday last?